src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py

```diff
@@ -0,0 +1,163 @@
+import json
+import pytest
+import requests
+from openeo_test_suite.lib.process_registry import ProcessData
+from openeo_test_suite.lib.process_selection import get_selected_processes
+from openeo_test_suite.lib.backend_under_test import (
+    get_backend_under_test,
+)
+import warnings
+
+
+@pytest.fixture(scope="module")
+def api_processes():
+    endpoint_path = "processes"
+    base_url = get_backend_under_test().connection.root_url
+    if not base_url:
+        raise ValueError("No backend URL configured")
+    if base_url.endswith("/"):
+        base_url = base_url[:-1]
+    full_endpoint_url = f"{base_url}/{endpoint_path}"
+    response = requests.get(full_endpoint_url)
+    if response.status_code != 200:
+        raise ValueError(
+            f"Failed to get processes from {full_endpoint_url}: {response.content}"
+        )
+    return json.loads(response.content)["processes"]
+
+
+def _get_test_id(val):
+    if isinstance(val, ProcessData):
+        return val.process_id
+
+
+@pytest.mark.parametrize(
+    "expected_process",
+    [p for p in get_selected_processes() if p.metadata],
+    ids=_get_test_id,
+)
+def test_process_metadata_functional(api_processes, expected_process, skipper):
+    """
+    Tests if the metadata of processes are correct, first tests if the process exists,
+    then tests if the parameters of processes are correct and finally tests if the return type of processes is correct.
+
+    Any process that has no metadata is skipped.
+
+    These are the functional parts of the process metadata e.g.: the parameters and return type.
+    """
+
+    skipper.skip_if_unsupported_process(expected_process.process_id)
+
+    actual_process = [
+        process
+        for process in api_processes
+        if process["id"] == expected_process.process_id
+    ][0]
+    # Tests if the parameters of processes are correct
+
+    expected_parameters = expected_process.metadata.get("parameters", [])
+    actual_parameters = actual_process["parameters"]
+
+    if len(expected_parameters) > len(actual_parameters):
+        warnings.warn(
```

Comment:
use a `logging` warning here instead of `warnings.warn` as the latter will not show up in the HTML report, while the former does

Reply:
Done.